backend/homework/7/src/main/java/com/example/inventory/VehicleInventory.java

```diff
@@ -0,0 +1,53 @@
+package com.example.inventory;
+
+import com.example.model.Vehicle;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * We have created vehicleInventory as Component and given scope as prototype so that it creates new instance every time
+ * it is been called.
+ */
+@Component
+@Scope("prototype")
+public class VehicleInventory {
+
+    private List<Vehicle> vehicles;
+
+    public VehicleInventory(){
+        this.vehicles = new ArrayList<>();
+    }
+
+    public void addVehicle(Vehicle vehicle){
+        vehicles.add(vehicle);
+    }
+
+    public List<Vehicle> getVehicle(){
+        return vehicles;
+    }
+
+    /**
+     * For finding the highest priced vehicle
+     * @return The vehicle with the highest price
+     */
+    public Vehicle findHighestPricedVehicle() {
+        return vehicles.stream()
```

Comment:
it is possible that the object will be null, so it's important to provide a proper message or handling mechanism for that scenario.